pkg/namespace/namespace.go

```diff
@@ -0,0 +1,142 @@
+package namespace
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/appuio/seiso/pkg/util"
+	"github.com/karrick/tparse/v2"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	core "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const cleanAnnotation = "syn.tools/clean"
+
+var (
+	resources = []schema.GroupVersionResource{
+		{Version: "v1", Resource: "pods"},
+		{Group: "apps", Version: "v1", Resource: "statefulsets"},
+		{Group: "apps", Version: "v1", Resource: "deployments"},
+		{Group: "apps", Version: "v1", Resource: "daemonsets"},
+		{Group: "extensions", Version: "v1beta1", Resource: "daemonsets"},
+		{Group: "extensions", Version: "v1beta1", Resource: "deployments"},
+	}
+)
+
+type (
+	NamespacesService struct {
+		configuration ServiceConfiguration
+		client        core.NamespaceInterface
+		dynamicClient dynamic.Interface
+		checkers      []Checker
+	}
+	ServiceConfiguration struct {
+		Batch bool
+	}
+	Checker interface {
+		NonEmptyNamespaces(context.Context, map[string]struct{}) error
+		Name() string
+	}
+)
+
+// NewNamespacesService creates a new Service instance
+func NewNamespacesService(client core.NamespaceInterface, dynamicClient dynamic.Interface, configuration ServiceConfiguration) NamespacesService {
+	return NamespacesService{
+		client:        client,
+		configuration: configuration,
+		checkers:      []Checker{NewHelmChecker(), NewResourceChecker(dynamicClient)},
+	}
+}
+
+func (nss NamespacesService) List(ctx context.Context, listOptions metav1.ListOptions) ([]corev1.Namespace, error) {
+	ns, err := nss.client.List(ctx, listOptions)
+	if err != nil {
+		return nil, err
+	}
+	return ns.Items, nil
+}
+
+func (nss NamespacesService) GetEmptyFor(ctx context.Context, namespaces []corev1.Namespace, duration string) ([]corev1.Namespace, error) {
+	now := time.Now()
+	emptyNamespaces := []corev1.Namespace{}
+	nonEmptyNamespaces := make(map[string]struct{}, len(namespaces))
+
+	for _, checker := range nss.checkers {
+		err := checker.NonEmptyNamespaces(ctx, nonEmptyNamespaces)
+		if err != nil {
+			return nil, fmt.Errorf("could not get %s resources %w", checker.Name(), err)
+		}
+	}
+
+	for _, ns := range namespaces {
+		if _, ok := nonEmptyNamespaces[ns.Name]; ok {
+			// Namespace is not empty
+			if _, ok := ns.Annotations[cleanAnnotation]; ok && !nss.configuration.Batch {
+				log.Warnf("Non empty namespace is annotated for deletion, skip: %q", ns.Name)
```

Comment:
The output is somewhat strange:
```console
INFO Seiso unknown, commit dirty, date today      
DEBU Using config                                  git="{0 . false version}" history="{3}" log="{debug false true}" namespace=default orphan="{1w ^[a-z0-9]{40}$}" resource="{[idunno=test] 1w 1s}"
DEBU Getting Namespaces                           
WARN Non empty namespace is annotated for deletion, skip: "seiso" 
INFO Showing results                               delete_after=1s
INFO Nothing found to be deleted.                 
```
I would reword it a bit:
```suggestion
				log.Warnf("Namespace is annotated for deletion, but not empty. Skipping %q", ns.Name)
```